File: examples/javascript/client_example.mjs
/** MembrainClient -- store and retrieve LLM memories with case-based reasoning. */

import { MembrainClient } from "membrain";

const client = new MembrainClient();

try {
  client.storeFact("Python was created by Guido van Rossum in 1991", 0.95);
  client.storePreference("user", "languages", "prefers strongly-typed languages", "strong");
  client.storeEntity("membrain", "software_project");
  client.storeConcept("Vector Similarity Search", "Finding closest vectors in high-dimensional space");

  client.storeCase(
    "User asked to summarize a 10-page document",
    "Split into sections, summarize each, then combine",
    "User satisfied with hierarchical summary",
    1.0,
  );

  console.log(`Stored ${client.count()} memories`);

  const results = client.search("programming languages", 5);
  console.log(`\nSearch results (${results.duration_ms}ms):`);
  for (const m of results.memories) {
    console.log(`  [${m.memory_type}] ${m.content.slice(0, 80)} (score=${m.score.toFixed(3)})`);
  }

  const filtered = client.search("vector search", 5, { memory_types: ["fact", "concept"] });
  console.log(`\nFiltered (facts+concepts): ${filtered.memories.length} results`);

  const cases = client.searchCases("summarize a long document", 3, 0.5);
  console.log(`\nCases: ${cases.positive_cases.length} positive, ${cases.negative_cases.length} negative`);
} finally {
  client.close();
}

File: examples/python/client_example.py
"""MembrainClient -- store and retrieve LLM memories with case-based reasoning."""

from membrain import MembrainClient


def main() -> None:
    with MembrainClient() as client:
        client.store_fact("Python was created by Guido van Rossum in 1991", confidence=0.95)
        client.store_preference(
            holder="user", subject="languages",
            preference="prefers strongly-typed languages", strength="strong",
        )
        client.store_entity(name="membrain", entity_type="software_project")
        client.store_concept(
            name="Vector Similarity Search",
            definition="Finding closest vectors in high-dimensional space",
        )

        client.store_case(
            problem="User asked to summarize a 10-page document",
            plan="Split into sections, summarize each, then combine",
            outcome="User satisfied with hierarchical summary",
            reward=1.0,
        )

        print(f"Stored {client.count()} memories")

        results = client.search("programming languages", limit=5)
        print(f"\nSearch results ({results.duration_ms}ms):")
        for memory in results.memories:
            print(f"  [{memory.memory_type}] {memory.content[:80]} (score={memory.score:.3f})")

        filtered = client.search("vector search", limit=5, filters={"memory_types": ["fact", "concept"]})
        print(f"\nFiltered (facts+concepts): {len(filtered.memories)} results")

        cases = client.search_cases("summarize a long document", limit=3, positive_reward_threshold=0.5)
        print(f"\nCases: {len(cases.positive_cases)} positive, {len(cases.negative_cases)} negative")


if __name__ == "__main__":
    main()
